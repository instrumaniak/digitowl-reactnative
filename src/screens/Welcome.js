/**
 *  Welcome screen shown when the user first opens the app
 */

import React from 'react'
import PropTypes from 'prop-types'

import {
  View,
  StatusBar,
  StyleSheet
} from 'react-native'

import {
  Button,
  Text
} from 'native-base'

import { themeColors } from '../config'

const Welcome = ({ navigation }) => (
  <View style={{flex: 1, backgroundColor: themeColors.brand }}>
    <StatusBar backgroundColor={ themeColors.brandDarken } />
    <View style={styles.titleContainer}>
      <Text style={styles.logoTitle}>DigitOwl</Text>
      <Text style={styles.logoSubTitle}>Expense tracker</Text>
    </View>
    <View style={styles.btnContainer}>
      <Button block light onPress={() => navigation.navigate('Login', {type: 'register'})}>
        <Text style={{color: themeColors.brand }} uppercase={false}>New to DigitOwl</Text>
      </Button>
      <Button block bordered light onPress={() => navigation.navigate('Login', {type: 'login'})}>
        <Text style={{color: 'white'}} uppercase={false}>Already using DigitOwl</Text>
      </Button>
    </View>
  </View>
)

Welcome.propTypes = {
  navigation: PropTypes.any
}

const styles = StyleSheet.create({
  titleContainer: {
    flex: 5,
    justifyContent: 'center',
    alignItems: 'center'
  },
  logoTitle: {
    fontSize: 35,
    marginBottom: 5,
    color: 'white'
  },
  logoSubTitle: {
    fontSize: 15,
    fontStyle: 'italic',
    color: 'white'
  },
  btnContainer: {
    flex: 1,
    justifyContent: 'space-around',
    padding: 10
  }
})

export default Welcome
