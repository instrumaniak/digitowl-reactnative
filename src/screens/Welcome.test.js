/**
 *   Tests for Welcome component
 */

import React from 'react'
import Welcome from './Welcome'
import renderer from 'react-test-renderer'

test('Renders correctly', () => {
  const tree = renderer.create(<Welcome />).toJSON()
  expect(tree).toMatchSnapshot()
})
