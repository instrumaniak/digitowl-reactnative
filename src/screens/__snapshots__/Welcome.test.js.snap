// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00b175",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 5,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "white",
            "fontSize": 35,
            "marginBottom": 5,
          },
        ]
      }
      uppercase={false}
    >
      DigitOwl
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "white",
            "fontSize": 15,
            "fontStyle": "italic",
          },
        ]
      }
      uppercase={false}
    >
      Expense tracker
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
        "padding": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#f4f4f4",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "#00b175",
            },
          ]
        }
        uppercase={false}
      >
        New to DigitOwl
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#f4f4f4",
          "borderLeftWidth": 1,
          "borderRadius": 5,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "elevation": null,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "white",
            },
          ]
        }
        uppercase={false}
      >
        Already using DigitOwl
      </Text>
    </View>
  </View>
</View>
`;
